src/test/java/com/cloudbees/plugins/credentials/CredentialsProviderTypeRestrictionTest.java

```diff
@@ -38,6 +43,10 @@ public void migrateDescriptorClassNameToID() {
 
         CredentialsProviderTypeRestriction.Includes includes = (CredentialsProviderTypeRestriction.Includes) includesRestriction;
         assertThat(includes.getType(), is("com.cloudbees.plugins.credentials.impl.DummyIdCredentials"));
+
+        CredentialsTypeFilter typeFilter = instance.getTypeFilter();
+        assertTrue(typeFilter.filter(ExtensionList.lookupSingleton(UsernamePasswordCredentialsImpl.DescriptorImpl.class)));
+        assertFalse(typeFilter.filter(ExtensionList.lookupSingleton(CertificateCredentialsImpl.DescriptorImpl.class)));
```

Comment:
Choosing to test the behavior of the exposed API rather than the implementation field.